File: 백준/Silver/11651. 좌표 정렬하기 2/좌표 정렬하기 2.java
import java.util.*;
import java.io.*;
public class Main{
    public static void main(String[] args) throws IOException{
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int N = Integer.parseInt(br.readLine());
        List<Integer[]> list = new ArrayList<>();
        for(int i = 0; i < N; i++){
            String[] numbers = br.readLine().split(" ");
            list.add(new Integer[]{Integer.parseInt(numbers[0]), Integer.parseInt(numbers[1])});
        }
        list.sort((a,b) -> {
            int cmp  = Integer.compare(a[1], b[1]);
            if(cmp == 0){
                return Integer.compare(a[0], b[0]);
            }
            return cmp;
        });
        for(int j = 0; j < N; j++){
            System.out.println(list.get(j)[0] + " " + list.get(j)[1]);
        }
    }
}

File: 백준/Silver/11651. 좌표 정렬하기 2/좌표 정렬하기 2.py
import sys
input = sys.stdin.readline
write = sys.stdout.write  # write를 간단하게 씀

n = int(input())
dots = [list(map(int, input().split())) for _ in range(n)]

dots.sort(key=lambda x: (x[1], x[0]))

for x, y in dots:
    write(f"{x} {y}\n")